ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBackupManager.java

```diff
@@ -54,51 +64,121 @@ public CloudBackupManager(VerifiableProperties properties, CloudConfig cloudConf
     super(replicationConfig, clusterMapConfig, storeKeyFactory, clusterMap, scheduler,
         virtualReplicatorCluster.getCurrentDataNodeId(), Collections.emptyList(), connectionPool, metricRegistry,
         requestNotification, storeKeyConverterFactory, transformerClassName);
-
-    CloudDestination cloudDestination = cloudDestinationFactory.getCloudDestination();
+    this.properties = properties;
+    this.cloudConfig = cloudConfig;
+    this.storeConfig = storeConfig;
+    this.virtualReplicatorCluster = virtualReplicatorCluster;
+    this.vcrMetrics = new VcrMetrics(metricRegistry);
+    this.cloudDestination = cloudDestinationFactory.getCloudDestination();
+    persistor =
+        new CloudTokenPersistor(replicaTokenFileName, mountPathToPartitionInfoList, replicationMetrics, clusterMap,
+            factory, cloudDestination);
+    try {
```

Comment:
It feels like most of this logic should be in start() rather than constructor.

Reply:
addressed.